src/common/str.c

```diff
@@ -252,6 +252,14 @@ int c11_string__cmp3(c11_string self, const char *other){
     return c11_string__cmp2(self, other, strlen(other));
 }
 
+bool c11_string_startswith(c11_string self, c11_string prefix) {
+    return strncmp(self.data, prefix.data, PK_MIN(self.size, prefix.size)) == 0;
+}
+
+bool c11_string_startswith_cstrn(c11_string self, const char *prefix, int size) {
```

Comment:
1. use double `__`
2. Use c11_string__startswith2.. c11_string__startswith3 for overloads

Reply:
In fact, c11_string is the same as a const char* + int size. So no need to add unnecessary overload.

Reply:
In fact, c11_string is the same as a const char* + int size. So no need to add unnecessary overload.